tooling/nargo/src/artifacts/debug_vars.rs

```diff
@@ -0,0 +1,75 @@
+use acvm::acir::brillig::Value;
+use noirc_printable_type::PrintableType;
+use std::collections::{HashMap, HashSet};
+
+#[derive(Debug, Default, Clone)]
+pub struct DebugVars {
+    id_to_name: HashMap<u32, String>,
+    active: HashSet<u32>,
+    id_to_value: HashMap<u32, Value>, // TODO: something more sophisticated for lexical levels
+    id_to_type: HashMap<u32, u32>,
+    types: HashMap<u32, PrintableType>,
+}
+
+impl DebugVars {
+    pub fn new(vars: &HashMap<u32, String>) -> Self {
+        let mut debug_vars = Self::default();
+        debug_vars.id_to_name = vars.clone();
+        debug_vars
+    }
+
+    pub fn get_variables<'a>(&'a self) -> Vec<(&'a str, &'a Value, &'a PrintableType)> {
+        self.active
+            .iter()
+            .filter_map(|var_id| {
+                self.id_to_name
+                    .get(var_id)
+                    .and_then(|name| self.id_to_value.get(var_id).map(|value| (name, value)))
+                    .and_then(|(name, value)| {
+                        self.id_to_type.get(var_id).map(|type_id| (name, value, type_id))
+                    })
+                    .and_then(|(name, value, type_id)| {
+                        self.types.get(type_id).map(|ptype| (name.as_str(), value, ptype))
+                    })
+            })
```

Comment:
```suggestion
                let name = self.id_to_name.get(var_id)?;
                let value = self.id_to_value.get(var_id)?;
                let type_id = self.id_to_type.get(var_id)?;
                let printable_type = self.types.get(type_id)?;

                Some((name.as_str(), value, printable_type))
```

We can avoid the usage of `and_then` with `?` which greatly helps readability imo.